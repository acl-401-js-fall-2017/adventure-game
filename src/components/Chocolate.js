import React, { Component } from 'react';
import chocPic from '../pics/chocolate.jpg';
import PropTypes from 'prop-types';


export default class Chocolate extends Component {
  constructor() {
    super();
    this.state = {
      chocMessage: 'You have avoided death, but another child was not so lucky',
    };
  }

  handleClick(value, inv = ''){
    const { handleInventory, inventory } = this.props;
    this.setState({
      message: value === 'correct' && inventory.length >= 1 ? this.state.chocMessage : 'The oompa loompa are about to sing about your death' 
    },() =>  {
      if (inv === 'loompa') {
        handleInventory('loompa ');
      }
    });
  }

  render() {
    
    return (
      <section className="challenge">
        <img src={chocPic} alt="" useMap="#chocMap" />
        <map name="chocMap" id="Map">
          <area data-inv="loompa" data-value="correct" alt="grid" shape="poly" coords="54,69,44,360,160,357,134,68" 
            onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
          <area alt="grid" shape="poly" coords="171,347,172,462,283,467,307,357" />
          <area alt="grid" shape="poly" coords="701,510,700,593,844,607,872,509" />
          <area data-inv="loompa" data-value="correct" alt="grid" shape="poly" coords="455,103,459,230,597,236,594,110" 
            onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)} />
        </map>
        <p className = "mission">
          The oompa loompas are coming after the kids, where are they going to come from?
        </p>
        <p className="message">
          Message: {this.state.message}
        </p>
      </section>
    );
  }

}

Chocolate.PropTypes = {
  message: PropTypes.string
};
