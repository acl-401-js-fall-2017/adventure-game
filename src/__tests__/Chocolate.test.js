import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Chocolate from '../components/Chocolate';

describe('Chocolate', () => {

  it('shows a chocolate', () => {
    const wrapper = shallow(<Chocolate handleInventory={() => {}} inventory={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});