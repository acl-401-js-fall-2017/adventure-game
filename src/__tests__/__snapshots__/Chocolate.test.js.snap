// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chocolate shows a chocolate 1`] = `
<section
  className="challenge"
>
  <img
    alt=""
    src="chocolate.jpg"
    useMap="#chocMap"
  />
  <map
    id="Map"
    name="chocMap"
  >
    <area
      alt="grid"
      coords="54,69,44,360,160,357,134,68"
      data-inv="loompa"
      data-value="correct"
      onClick={[Function]}
      shape="poly"
    />
    <area
      alt="grid"
      coords="171,347,172,462,283,467,307,357"
      shape="poly"
    />
    <area
      alt="grid"
      coords="701,510,700,593,844,607,872,509"
      shape="poly"
    />
    <area
      alt="grid"
      coords="455,103,459,230,597,236,594,110"
      data-inv="loompa"
      data-value="correct"
      onClick={[Function]}
      shape="poly"
    />
  </map>
  <p
    className="mission"
  >
    The oompa loompas are coming after the kids, where are they going to come from?
  </p>
  <p
    className="message"
  >
    Message: 
  </p>
</section>
`;
